src/components/settings/NumberSetting.tsx

```diff
@@ -62,41 +65,50 @@ export const NumberSetting = ({
     handleUpdate,
     showSlider,
     disabled = false,
+    handleLiveUpdate,
     listItemTextSx: sx,
 }: Props) => {
     const { t } = useTranslation();
 
     const [isDialogOpen, setIsDialogOpen] = useState(false);
     const [dialogValue, setDialogValue] = useState(value);
+    const [originalValue, setOriginalValue] = useState(value);
 
-    const closeDialog = useCallback(
-        (resetValue: boolean) => {
-            setIsDialogOpen(false);
-
-            if (resetValue) {
-                setDialogValue(value);
+    const updateValue = useCallback(
+        (newValue: number) => {
+            setDialogValue(newValue);
+            if (handleLiveUpdate) {
+                handleLiveUpdate(newValue);
             }
         },
-        [value],
+        [handleLiveUpdate],
     );
 
-    const closeDialogWithReset = useCallback(() => closeDialog(true), [closeDialog]);
+    const cancel = useCallback(() => {
```

Comment:
`cancel` and  `resetToDefault` are basically the same function, should be merged

Reply:
They're similar, but not the same. They call the same functions, but the values and conditions are different. Do you have any recommendations about they should be combined?

Reply:
you could do a function that gets passed an optional value to reset to as an argument